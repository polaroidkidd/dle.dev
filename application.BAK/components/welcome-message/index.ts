import { WelcomeMessage } from "./welcome-message";

export { WelcomeMessage };