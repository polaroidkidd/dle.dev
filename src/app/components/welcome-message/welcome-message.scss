@import "../../styles/main";

.welcome-message {
  display: flex;
  justify-content: center;
  text-align: center;
  @media (max-width: 768px) {
    font-size: 2.5rem;
    font-weight: 200;
  }
}

.additional-paragraph-styles {
  margin-top: 1rem;
  text-align: center;
}