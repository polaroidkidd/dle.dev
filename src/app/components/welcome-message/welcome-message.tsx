import React from 'react';
import './welcome-message.scss';
import {NavLink} from 'react-router-dom';
import {ROUTES} from '../navigation';

const WelcomeMessage: React.FC = () => {
  return (
    <>
      <section>
        <h2 className="heading-secondary welcome-message">
          Hi! I'm Daniel Einars.
        </h2>
        <h3 className="heading-tertiary u-center-text__horizontal u-center-text__padding">
          I build websites & servers.
        </h3>

        <h4 className="heading-quaternary u-center-text__horizontal">
          Check out my <NavLink to={ROUTES.BLOG}>first blog entry</NavLink> on
          using React with Typescript!
        </h4>
      </section>
    </>
  );
};
export {WelcomeMessage};
