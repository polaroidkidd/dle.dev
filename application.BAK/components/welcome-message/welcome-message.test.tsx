import React from "react";
import renderer from "react-test-renderer";
import { WelcomeMessage } from "./welcome-message";
import { createBrowserHistory } from "history";
import { Router } from "react-router-dom";


test("Render WelcomeMessage with text", () => {

  const history = createBrowserHistory();

  const tree = renderer.create(
    <Router history={history}>
      <WelcomeMessage/>
    </Router>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});