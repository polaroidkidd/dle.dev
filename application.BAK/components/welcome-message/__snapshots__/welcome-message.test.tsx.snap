// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render WelcomeMessage with text 1`] = `
<section>
  <h2
    className="heading-secondary welcome-message"
  >
    Hi! I'm Daniel Einars.
  </h2>
  <h3
    className="heading-tertiary u-center-text__horizontal u-center-text__padding"
  >
    I build websites & servers.
  </h3>
  <h4
    className="heading-quaternary u-center-text__horizontal"
  >
    Check out my 
    <a
      aria-current={null}
      href="/blog"
      onClick={[Function]}
    >
      first blog entry
    </a>
     on using React with Typescript!
  </h4>
</section>
`;
